src/osd/osd_types.cc

```diff
@@ -1107,72 +1107,39 @@ std::string pg_vector_string(const vector<int32_t> &a)
 std::string pg_state_string(uint64_t state)
 {
   CachedStackStringStream css;
-  if (state & PG_STATE_STALE)
-    *css << "stale+";
-  if (state & PG_STATE_CREATING)
-    *css << "creating+";
-  if (state & PG_STATE_ACTIVE)
-    *css << "active+";
-  if (state & PG_STATE_ACTIVATING)
-    *css << "activating+";
-  if (state & PG_STATE_CLEAN)
-    *css << "clean+";
-  if (state & PG_STATE_RECOVERY_WAIT)
-    *css << "recovery_wait+";
-  if (state & PG_STATE_RECOVERY_TOOFULL)
-    *css << "recovery_toofull+";
-  if (state & PG_STATE_RECOVERING)
-    *css << "recovering+";
-  if (state & PG_STATE_FORCED_RECOVERY)
-    *css << "forced_recovery+";
-  if (state & PG_STATE_DOWN)
-    *css << "down+";
-  if (state & PG_STATE_RECOVERY_UNFOUND)
-    *css << "recovery_unfound+";
-  if (state & PG_STATE_BACKFILL_UNFOUND)
-    *css << "backfill_unfound+";
-  if (state & PG_STATE_UNDERSIZED)
-    *css << "undersized+";
-  if (state & PG_STATE_DEGRADED)
-    *css << "degraded+";
-  if (state & PG_STATE_REMAPPED)
-    *css << "remapped+";
-  if (state & PG_STATE_PREMERGE)
-    *css << "premerge+";
-  if (state & PG_STATE_SCRUBBING)
-    *css << "scrubbing+";
-  if (state & PG_STATE_DEEP_SCRUB)
-    *css << "deep+";
-  if (state & PG_STATE_INCONSISTENT)
-    *css << "inconsistent+";
-  if (state & PG_STATE_PEERING)
-    *css << "peering+";
-  if (state & PG_STATE_REPAIR)
-    *css << "repair+";
-  if (state & PG_STATE_BACKFILL_WAIT)
-    *css << "backfill_wait+";
-  if (state & PG_STATE_BACKFILLING)
-    *css << "backfilling+";
-  if (state & PG_STATE_FORCED_BACKFILL)
-    *css << "forced_backfill+";
-  if (state & PG_STATE_BACKFILL_TOOFULL)
-    *css << "backfill_toofull+";
-  if (state & PG_STATE_INCOMPLETE)
-    *css << "incomplete+";
-  if (state & PG_STATE_PEERED)
-    *css << "peered+";
-  if (state & PG_STATE_SNAPTRIM)
-    *css << "snaptrim+";
-  if (state & PG_STATE_SNAPTRIM_WAIT)
-    *css << "snaptrim_wait+";
-  if (state & PG_STATE_SNAPTRIM_ERROR)
-    *css << "snaptrim_error+";
-  if (state & PG_STATE_FAILED_REPAIR)
-    *css << "failed_repair+";
-  if (state & PG_STATE_LAGGY)
-    *css << "laggy+";
-  if (state & PG_STATE_WAIT)
-    *css << "wait+";
+  if (state & PG_STATE_ACTIVATING) *css << "activating+";
+  if (state & PG_STATE_ACTIVE) *css << "active+";
+  if (state & PG_STATE_BACKFILLING) *css << "backfilling+";
+  if (state & PG_STATE_BACKFILL_TOOFULL) *css << "backfill_toofull+";
+  if (state & PG_STATE_BACKFILL_UNFOUND) *css << "backfill_unfound+";
+  if (state & PG_STATE_BACKFILL_WAIT) *css << "backfill_wait+";
+  if (state & PG_STATE_CLEAN) *css << "clean+";
+  if (state & PG_STATE_CREATING) *css << "creating+";
+  if (state & PG_STATE_DEEP_SCRUB) *css << "deep+";
```

Comment:
I'm not sure we want do this entirely, things like "scrubbing+deep" make more sense together

Reply:
Hm, we could merge them somehow as a proxy dual state "deep_scrubbing". Would you ever have "deep" and not "scrubbing"?

Reply:
@batrick @neha-ojha This is not a good idea, because script and test might parse output which will become broken if the pg states are rearranged.